.github/workflows/main.yml

```diff
@@ -6,42 +6,72 @@ on:
   pull_request:
   workflow_dispatch:
 
+concurrency:
```

Comment:
Support canceling in-progress workflow runs when a pull request is rapidly updated. All other runs behave as always.